app/src/main/kotlin/com/adesso/movee/scene/profile/ProfileViewModel.kt

```diff
@@ -32,12 +32,8 @@ class ProfileViewModel @Inject constructor(
     }
 
     private fun getLoginState() {
-        bgScope.launch {
-            val loginStateResult = getLoginStateUseCase.run(UseCase.None)
-
-            onUIThread {
-                loginStateResult.either(::handleFailure, ::handleLoginStateSuccess)
-            }
+        getLoginStateUseCase.invoke(viewModelScope, UseCase.None) {
+            it.either(::handleFailure, ::handleLoginStateSuccess)
```

Comment:
handleLoginStateSuccess() calls postLoginState() inside. postLoginState() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postLoginState(). Also, handleLoginStateSuccess() calls updateUserDetails() which is a network operation and switches context with Dispatchers.IO. Using postValue() at postLoginState() would better suit here.